crates/adapters/src/integrated/delta_table/output.rs

```diff
@@ -0,0 +1,515 @@
+                                    false,
+                                )))
+                                .await;
+                        }
+                    }
+                }
+                Some(Command::BatchEnd) => {
+                    trace!(
+                        "delta_table {}: finished writing output records to '{batch_uri}'",
+                        &inner.endpoint_name
+                    );
+                    if let Some(_writer) = writer.take() {
+                        let _ = response_sender.send(Ok(())).await;
+                        // TODO: commit transaction.
+                    } else {
+                        error!(
+                            "delta_table {}: received a BatchEnd without a matching BatchStart",
+                            &inner.endpoint_name
+                        );
+                    }
+                }
+                Some(Command::Data(batch)) => {
+                    if let Some(writer) = &mut writer {
+                        let num_rows = batch.num_rows();
+                        trace!(
+                            "delta_table {}: writing {num_rows} records to '{batch_uri}'",
+                            &inner.endpoint_name
+                        );
+
+                        match writer.write(batch).await {
+                            Ok(()) => {
+                                let num_bytes = writer.buffer_len();
+
+                                match writer.flush().await {
+                                    Ok(_) => {
+                                        if let Some(controller) = inner.controller.upgrade() {
+                                            controller.status.output_buffer(
+                                                inner.endpoint_id,
+                                                num_bytes,
+                                                num_rows,
+                                            )
+                                        };
+
+                                        let _ = response_sender.send(Ok(())).await;
+                                    }
+                                    Err(e) => {
+                                        let _ = response_sender
+                                            .send(Err((
+                                                anyhow!(
+                                                    "error writing parquet file '{batch_uri}': {e}"
+                                                ),
+                                                false,
+                                            )))
+                                            .await;
+                                    }
+                                }
+                            }
+                            Err(e) => {
+                                let _ = response_sender.send(Err((anyhow!("error writing parquet file with {num_rows} rows to '{batch_uri}': {e}"), false))).await;
```

Comment:
it's nice to report errors, but I have no idea what happens to the system when one of these hits
maybe a new file should be started?

Reply:
It does start a new file. We can add retries, but if that fails your only options are to live with incomplete data or rebuild the output from scratch (which we don't currently support).  I'll add a TODO to add logic for retries.